test/stream.test.ts

```diff
@@ -356,11 +373,46 @@ describe('DataAndMoneyStream', function () {
       })
     })
 
-    it('should not allow more data to be written once the stream is closed', async function () {
+    it('should not allow more data to be written once the stream is closed and throw an error', async function () {
       const clientStream = this.clientConn.createStream()
       clientStream.end()
       assert.throws(() => clientStream.write('hello'), 'write after end')
     })
+
+    it('should not allow more data to be written once the stream is closed and emit an error', async function () {
+      const clientStream = this.clientConn.createStream()
+      clientStream.on('error', (err: Error) => {
```

Comment:
If it's also throwing it might be better not to emit an error. Node will complain if there are unhandled error events, which are kind of a pain to deal with